frontend/src/components/projectDetail/tests/favorites.test.js

```diff
@@ -0,0 +1,24 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import '@testing-library/jest-dom';
+
+import { ReduxIntlProviders } from '../../../utils/testWithIntl';
+import { AddToFavorites } from '../favorites';
+
+describe('AddToFavorites button', () => {
+  it('renders button when project id = 1', async () => {
+    const props = {
+      projectId: 1,
+    };
+    const { container } = render(
+      <ReduxIntlProviders>
+        <AddToFavorites {...props} />
+      </ReduxIntlProviders>,
+    );
+    const button = screen.getByRole('button');
+    expect(button.className).toBe(' input-reset base-font bg-white blue-dark f6 bn pointer');
+    expect(button.className).not.toBe('dn input-reset base-font bg-white blue-dark f6 bn pointer');
+    expect(container.querySelector('svg').classList.value).toBe('pt3 pr2 v-btm o-50 ');
+    expect(button.textContent).toBe('Add to Favorites');
```

Comment:
We could mock the API request and test the click on the button.

Reply:
@d-rita in case you have not started working on that change, I think we can dedicate time to other components instead of this one, as this functionality doesn't seem to be very widely used.